assets/js/src/utils/index.js

```diff
@@ -26,6 +26,23 @@ export const getProductFieldObject = ( product, quantity ) => {
 		),
 		...variantData,
 	};
+
+	// Apply discounts to ecommerce events.
+	// https://developers.google.com/analytics/devguides/collection/ga4/apply-discount?client_type=gtag
+	if (
+		typeof product.price_after_coupon_discount === 'number' &&
+		! isNaN( product.price_after_coupon_discount ) &&
+		product.price_after_coupon_discount !== product.prices.price
+	) {
+		data.discount =
+			product.prices.price - product.price_after_coupon_discount;
```

Comment:
We need to wrap this with formatPrice() as well. Otherwise the discount is send without decimals.